README.md

```diff
@@ -344,7 +344,7 @@ You can also define multiple files in `config/samples`, but you must specify the
    - `beta`: (Optional) Determines whether ModuleTemplate must have the beta flag or not. The type is bool.
    - `labels`: (Optional) Additional labels for ModuleTemplate.
    - `annotations`: (Optional) Additional annotations for ModuleTemplate.
-   - `customStateCheck`: (Optional) Specifies custom state checks for the module.
+   - `customStateCheck`: (Optional) DEPRECATED, Specifies custom state checks for the module.
```

Comment:
```suggestion
   - `customStateCheck`: (Optional) [DEPRECATED] Specifies custom state checks for the module.
   
   > **CAUTION:** This field is deprecated and will be removed on {DATE}. 
```

Reply:
or we could provide both the deprecation and deletion dates.

```suggestion
   - `customStateCheck`: (Optional) [DEPRECATED], Specifies custom state checks for the module.

   > **CAUTION:** This field was deprecated at the end of July 2024 and will be removed on {DATE}. 
```

Reply:
I think we don't know the deletion date/API version. So I'd rather skip this.